src/callbacks.rs

```diff
@@ -36,61 +43,62 @@ impl SelfCallbacks for Contract {
                 if is_final {
                     remove_indices.push(index);
                 }
-                total_balance += claim_amount.0;
+                total_balance = total_balance.saturating_add(claim_amount);
                 let event = FtLockupClaimLockup {
                     id: index,
                     amount: claim_amount,
                 };
                 events.push(event);
```

Comment:
This could lead to confusion or vulnerability. Let's use `checked_*` instead of `saturating_*`